flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -608,6 +610,33 @@ public String getValue(ConfigOption<?> configOption) {
 		return o == null ? null : o.toString();
 	}
 
+	/**
+	 * Returns the value associated with the given config option as an enum.
+	 *
+	 * @param enumClass    The return enum class
+	 * @param configOption The configuration option
+	 * @throws IllegalArgumentException If the string associated with the given config option cannot
+	 *                                  be parsed as a value of the provided enum class.
+	 */
+	@PublicEvolving
+	public <T extends Enum<T>> T getAsEnum(
```

Comment:
nit: This would be more consistent with other getters.
```suggestion
	public <T extends Enum<T>> T getEnum(
```

Reply:
I think you are right. It differs a bit though in the sense that `ConfigOption<Enum>` is not possible. On the other hand there is `getBytes`

Reply:
I see... then I am ok with both actually.